File: app/src/main/java/com/cn29/aac/ui/shopping/vm/AlbumFragmentViewModelFactory.java
package com.cn29.aac.ui.shopping.vm;

import com.cn29.aac.repo.itunes.ItunesRepository;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

/**
 * Created by Charles Ng on 20/10/2017.
 */

public class AlbumFragmentViewModelFactory extends ViewModelProvider.NewInstanceFactory {

  private ItunesRepository itunesRepository;

  @Inject
  public AlbumFragmentViewModelFactory(ItunesRepository itunesRepository) {
    super();
    this.itunesRepository = itunesRepository;
  }

  @NonNull
  @Override
  public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
    return (T) new AlbumFragmentViewModel(this.itunesRepository);
  }
}

File: app/src/main/java/com/cn29/aac/ui/shopping/vm/AlbumFragmentViewModelFactory.kt
package com.cn29.aac.ui.shopping.vm

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import com.cn29.aac.repo.itunes.ItunesRepository
import javax.inject.Inject

/**
 * Created by Charles Ng on 20/10/2017.
 */
class AlbumFragmentViewModelFactory @Inject constructor(private val itunesRepository: ItunesRepository) :
        NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return AlbumFragmentViewModel(itunesRepository) as T
    }

}